### ISSUE TITLE
Cannot invoke action with "enter" key on first menu item of submenu after opening submenu with "enter" key.

### ISSUE THREAD

**Issue by Peter-Schorn** [adobe/react-spectrum#9686]
### Provide a general summary of the issue here

For the `Menu` component with a submenu, when you open the submenu by pressing the "enter" key, pressing the "enter" key again does NOT invoke the action on the first submenu item (which is automatically focused when the submenu opens).

### 🤔 Expected Behavior?

After the submenu is opened by pressing the "enter" key, pressing the "enter" key again should invoke the action for the first submenu item (which is automatically focused when the submenu opens).

### 😯 Current Behavior

Pressing the "enter" key does NOT invoke the action on the submenu item. However, if you navigate to a different menu item and then back to the previous menu item using arrow keys, pressing the "enter" key does invoke the action on the menu item.

### 🖥️ Steps to Reproduce

I originally discovered this issue by experimenting with the [submenu](https://react-aria.adobe.com/Menu#submenus) example code. If you open the "Share" submenu with the "enter" key and then press the "enter" key again, the action for the "SMS" menu item will NOT be invoked (even though it is focused) and the submenu will stay open. The following stackblitz is forked from that example code.

## Steps to Reproduce

* Open [this](https://stackblitz.com/edit/14xxmfky?file=src%2FExample.tsx) stackblitz project.
* Click to open the "actions" menu.
* Use the arrow keys to focus the "Share" menu item.
* Press the "enter" key to open the submenu of the "Share" menu item.
* Press the "enter" key again. This should invoke the action for the "SMS" menu item. It doesn't.

The following screen recording demonstrates the bug.

https://github.com/user-attachments/assets/6ac4db58-4f93-4b47-b006-274c7873baab

### Version

react-aria-components: 1.15.1

### What browsers are you seeing the problem on?

Chrome, Firefox, Safari

### What operating system are you using?

macOS

**Comment by LFDanLu** [comment]
Thanks for reporting this, I've reproduced this locally. Looks like focus is being moved to the submenu rather than the submenu item when pressing Enter which is definitely a bug, perhaps happening due to the double render that needs to happen to populate the menu's contents. Will have to dig some more for a fix, perhaps we could have the Enter event be propagated to the tracked focused key or fix this focus marshaling issue.

**Comment by Peter-Schorn** [comment]
Please note that opening the submenu with a right arrow key and then pressing the "enter" key does invoke the action on the first submenu item, as expected.

**Comment by LFDanLu** [comment]
Good point about that, it seems to something about the tracked interaction modality due to changes we made in https://github.com/adobe/react-spectrum/pull/8327. Right now Enter fires a programatic click on the menu item changing our modality to `virtual` instead of `keyboard`, messing up some of the `focusSafely` logic. I opened a PR to fix: https://github.com/adobe/react-spectrum/pull/9691
